### DISCUSSION TITLE
Error in Mssage update event

### DISCUSSION THREAD

**Discussion by end3r-man** [discord-php/DiscordPHP#1238] [Q&A] [answered]
`public function handle(Message $message, Discord $discord, ?Message $oldMessage)
    {
        echo "Old {$oldMessage->content}, message {$message->content}";
    }`

` Attempt to read property "content" on null

  at app/Events/MessageEdit.php:24
     20▕      * Handle the event.
     21▕      */
     22▕     public function handle(Message $message, Discord $discord, ?Message $oldMessage)
     23▕     {
  ➜  24▕         echo "Old {$oldMessage->content}, message {$message->content}";
     25▕     }
     26▕ }
     27▕ 

  1   app/Events/MessageEdit.php:24
      Illuminate\Foundation\Bootstrap\HandleExceptions::Illuminate\Foundation\Bootstrap\{closure}()
      +38 vendor frames`

**Comment by valzargaming** [comment, marked as answer]
Check if $oldMessage is set. `?Message` is the same as `Message|null`. You're accessing a property on an object that doesn't exist.

**Reply by civilCornball** [reply]
It'd be something like...
```
if (is_null($oldMessage))
{
echo "No old message detected.";
return;
}
```
and you wouldn't try to retrieve the $oldMessage->content in the same line as the $message->content.

**Reply by Log1x** [reply]
FWIW `$oldMessage` depends on DiscordPHP's cache and is not something returned by the Discord API. If you want proper update/delete logging – you will have to store/compare messages yourself.

**Reply by end3r-man** [reply]
> FWIW `$oldMessage` depends on DiscordPHP's cache and is not something returned by the Discord API. If you want proper update/delete logging – you will have to store/compare messages yourself.

tnx man
